src/EnergyPlus/Autosizing/HeatingWaterDesAirInletTempSizing.cc

```diff
@@ -0,0 +1,142 @@
+//
+// (3) Neither the name of the University of California, Lawrence Berkeley National Laboratory,
+//     the University of Illinois, U.S. Dept. of Energy nor the names of its contributors may be
+//     used to endorse or promote products derived from this software without specific prior
+//     written permission.
+//
+// (4) Use of EnergyPlus(TM) Name. If Licensee (i) distributes the software in stand-alone form
+//     without changes from the version obtained under this License, or (ii) Licensee makes a
+//     reference solely to the software portion of its product, Licensee must refer to the
+//     software as "EnergyPlus version X" software, where "X" is the version number Licensee
+//     obtained under this License and may not use a different name for the software. Except as
+//     specifically required in this Section (4), Licensee shall not use in a company name, a
+//     product name, in advertising, publicity, or other promotional activities any name, trade
+//     name, trademark, logo, or other designation of "EnergyPlus", "E+", "e+" or confusingly
+//     similar designation, without the U.S. Department of Energy's prior written consent.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS OR
+// IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY
+// AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR
+// CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+// CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+// SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+// THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR
+// OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+// POSSIBILITY OF SUCH DAMAGE.
+
+#include <EnergyPlus/Autosizing/HeatingWaterDesAirInletTempSizing.hh>
+#include <EnergyPlus/DataAirSystems.hh>
+#include <EnergyPlus/DataEnvironment.hh>
+#include <EnergyPlus/DataSizing.hh>
+#include <EnergyPlus/UtilityRoutines.hh>
+#include <EnergyPlus/api/TypeDefs.h>
+
+#include <EnergyPlus/ReportSizingManager.hh>
+
+namespace EnergyPlus {
+
+void HeatingWaterDesAirInletTempSizer::initializeWithinEP(EnergyPlusData &state,
+                                                          std::string const &_compType,
+                                                          std::string const &_compName,
+                                                          bool const printWarningFlag)
+{
+    BaseSizer::initializeWithinEP(state, _compType, _compName, printWarningFlag);
+    this->sizingString = "Rated Inlet Air Temperature";
+    this->termUnitFinalZoneSizing = DataSizing::TermUnitFinalZoneSizing;
+    this->totalSystemAirVolumeFlowRate = DataSizing::DataFlowUsedForSizing;
```

Comment:
This is much cleaner now. And shows what these sizing types need over the base class. I wonder how small the base initializer will wind up being or if there are some sizers that only need minimal info which would make the initializer moot.